api/envoy/config/route/v3/route_components.proto

```diff
@@ -1585,7 +1585,7 @@ message RateLimit {
 //   value.
 //
 //  [#next-major-version: HeaderMatcher should be refactored to use StringMatcher.]
-// [#next-free-field: 12]
+// [#next-free-field: 13]
```

Comment:
Did you consider instead to make `StringMatcher` an optional way to specify path, header, etc? This is what we'd like to unify behind.

Reply:
You mean internally use StringMatcher to do the matching instead of using Envoy::Http::HeaderMatcher/Envoy::Http::HeaderData

Reply:
Optimally I agree with @htuch. Could we deprecate exact_match, safe_regex_match, prefix_match, and suffix_match and instead add a string_match?